PokemonAPI/PokemonAPI.BusinessLayer/Validator/TrainerValidator.cs

```diff
@@ -0,0 +1,48 @@
+﻿using FluentValidation;
+using PokemonAPI.DomainLayer.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace PokemonAPI.BusinessLayer.Validator
+{
+    public class TrainerValidator : AbstractValidator<TrainerDTO>
+    {
+		public TrainerValidator()
+		{
+			RuleFor(x => x.FullName).Cascade(CascadeMode.StopOnFirstFailure).NotEmpty().WithMessage("Trainer full name is required")
+									.Must(fullName => IsNameAndSurname(fullName)).WithMessage("Full name must contain name and surname")
+									.Must(fullName => AllLetters(fullName)).WithMessage("Full name must contain only letters");
+		}
+
+		public bool AllLetters(string fullName)
+        {
+			string[] splitedFullName = fullName.Split(' ');
+			var onlyLetters = true;
+
+			foreach (var name in splitedFullName)
+            {
+				if (!name.All(Char.IsLetter))
+                {
+					onlyLetters = false;
```

Comment:
You can just `return false;` here because this is rule that breaks this loop ie no need to further go check all the letters when you just validated.. that all letters are in fact **not** letters